express_relay/programs/express_relay/src/utils.rs

```diff
@@ -30,26 +30,19 @@ pub fn transfer_lamports_cpi<'info>(
     amount: u64,
     system_program: AccountInfo<'info>,
 ) -> Result<()> {
-    let cpi_accounts = Transfer {
-        from: from.clone(),
-        to: to.clone(),
-    };
+    if amount > 0 {
+        let cpi_accounts = Transfer {
+            from: from.clone(),
+            to: to.clone(),
+        };
 
-    transfer(CpiContext::new(system_program, cpi_accounts), amount)?;
-
-    Ok(())
-}
-
-pub fn validate_pda(pda: &Pubkey, program_id: &Pubkey, seeds: &[&[u8]]) -> Result<()> {
-    let (pda_calculated, _) = Pubkey::find_program_address(seeds, program_id);
-    if pda != &pda_calculated {
-        return err!(ErrorCode::InvalidPDAProvided);
+        transfer(CpiContext::new(system_program, cpi_accounts), amount)?;
     }
 
     Ok(())
 }
 
-pub fn num_permissions_in_tx(sysvar_instructions: UncheckedAccount, permission: Option<Pubkey>, protocol: Option<Pubkey>) -> Result<u16> {
+pub fn num_permissions_in_tx(sysvar_instructions: UncheckedAccount, permission: Option<Pubkey>, router: Option<Pubkey>) -> Result<u16> {
```

Comment:
I think this function interface is a bit shaky. We have 2 usecases, one is to make sure there is only 1 instruction and the other is to make sure that it's with a specific permission and router. It should not be possible to provide only permission or only router since it doesn't make a lot of sense.